modules/db/src/main/scala/gem/dao/composite/Coordinates.scala

```diff
@@ -4,19 +4,28 @@
 package gem.dao.composite
 
 import doobie._
-import gem.math.{ Angle, HourAngle, Coordinates, Declination, RightAscension }
+import gem.dao.meta._
+
+import gem.math._
 
 trait CoordinatesComposite {
 
-  /** Map Coordinates as a (µsec, µasec) pair. */
-  implicit val CompositeCoordinates: Composite[Coordinates] =
-    Composite[(Long, Long)].imap {
-      case (ra, dec) =>
-        Coordinates(
-          RightAscension(HourAngle.fromMicroseconds(ra)),
-          Declination.unsafeFromAngle(Angle.fromMicroarcseconds(dec))
-        )
-      }(c => (c.ra.toHourAngle.toMicroseconds, c.dec.toAngle.toMicroarcseconds))
+  /** Coordinates composite, laid out in natural order, in microarcseconds. */
+  implicit val CoordinatesComposite: Composite[Coordinates] =
+    CoordinatesCompositeLemmas.CoordinatesComposite
+
+  implicit val CoordinatesOptionComposite: Composite[Option[Coordinates]] =
+    CoordinatesCompositeLemmas.CoordinatesOptionComposite
 
 }
 object CoordinatesComposite extends CoordinatesComposite
+
+/** Derivation of Composite instances for Coordinates. */
+private object CoordinatesCompositeLemmas {
```

Comment:
The issue here is that we're deriving the instance implicitly but also want to *export* it implicitly, which we can't do without causing an infinite loop. So we're pushing them into a helper object and delegating to them from `CoordinatesComposite` above.

Reply:
If you can type `implicitly` and summons an instance below, why do we need to export an implicit value in `CoordinatesComposite`?

Reply:
Yeah, if you import the RA and Dec instances you can then get the Coordinates one for free. I just wanted to provide it as a direct import. The problem is that you can get an error like `can't find Composite[ComplexType]` and you don't know which part of it is missing and you have to go through this debugging process where you break the type up try to figure out which particular part isn't available. So I was trying to avoid that by providing composites for bigger types which means fewer imports in client code (and probably faster compilation because the instances are derived once). But I don't feel strongly about it. We can get rid of this and add the necessary "smaller" instances at the call sites.

Reply:
No, no need to change anything.  That makes sense to me.